Sources/Classes/core/Storage/MiniAppSecureStorage.swift

```diff
@@ -0,0 +1,256 @@
+        try secureStorageData.write(to: secureStoragePath, options: .completeFileProtectionUnlessOpen)
+        MiniAppLogger.d("🔑 Secure Storage: write store to disk completed")
+    }
+
+    // MARK: - Clear
+    static func clearSecureStorage() throws {
+        MiniAppLogger.d("🔑 Secure Storage: destroy")
+        let cachePath = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask)[0]
+        let miniAppPath = cachePath.appendingPathComponent("/MiniApp/")
+        guard let contentNames = try? FileManager.default.contentsOfDirectory(atPath: miniAppPath.path) else { return }
+        for name in contentNames {
+            let url = miniAppPath.appendingPathComponent("/" + name)
+            if let isDirectory = (try url.resourceValues(forKeys: [.isDirectoryKey])).isDirectory, isDirectory {
+                do {
+                    try FileManager.default.removeItem(at: url.appendingPathComponent("/" + storageFullName))
+                    MiniAppLogger.d("🔑 Secure Storage: destroyed storaged for \(name)")
+                } catch {
+                    MiniAppLogger.d("🔑 Secure Storage: could not destroy storaged for \(name)")
+                }
+            } else {
+                MiniAppLogger.d("🔑 Secure Storage: ignored \(name)")
+            }
+        }
+    }
+
+    public static func clearSecureStorage(for miniAppId: String) throws {
+        MiniAppLogger.d("🔑 Secure Storage: destroy")
+        try FileManager.default.removeItem(at: storagePath(appId: miniAppId))
+    }
+
+    public func size() throws -> UInt64 {
+        let fileSize = MiniAppSecureStorage.storagePath(appId: appId).fileSize
+        guard fileSize > 0 else { throw MiniAppSecureStorageError.storageFileEmpty }
+        MiniAppLogger.d("🔑 Secure Storage: size -> \(fileSize)")
+        return fileSize
+    }
+
+    public static func storageSize(for miniAppId: String) throws -> UInt64 {
+        let fileSize = MiniAppSecureStorage.storagePath(appId: miniAppId).fileSize
+        guard fileSize > 0 else { throw MiniAppSecureStorageError.storageFileEmpty }
+        MiniAppLogger.d("🔑 Secure Storage: size -> \(fileSize)")
+        return fileSize
+    }
+}
+
+struct MiniAppSecureStorageSize: Codable {
+    let used: UInt64
+    let max: UInt64
+}
+
+public enum MiniAppSecureStorageError: Error, MiniAppErrorProtocol, Equatable {
+    case unknown(description: String)
+    case storageNotExistent
+    case storageFileEmpty
+    case storageBusyProcessing
+    case storageKeyNotFound
+
+    var name: String {
+        switch self {
+        case .unknown:
+            return "unknown"
+        case .storageNotExistent:
+            return "storageNotExistent"
+        case .storageFileEmpty:
+            return "storageFileEmpty"
+        case .storageBusyProcessing:
+            return "storageBusyProcessing"
+        case .storageKeyNotFound:
+            return "storageKeyNotFound"
```

Comment:
The `name` needs to match what's defined on the JS side so that it can be parsed to the correct error type (but I don't think @leorjoseph has added the error types yet in JS).

The names I suggested are `SecureStorageFullError` and `SecureStorageIOError` but maybe we also need some others.

Reply:
ok sounds good I'll wait for the specs